source/extensions/filters/http/common/aws/signer_impl.h

```diff
@@ -0,0 +1,78 @@
+#pragma once
+
+#include "common/common/logger.h"
+#include "common/common/utility.h"
+
+#include "extensions/filters/http/common/aws/credentials_provider.h"
+#include "extensions/filters/http/common/aws/region_provider.h"
+#include "extensions/filters/http/common/aws/signer.h"
```

Comment:
Would it make sense to have an abstract signing API in include/envoy somewhere, and have AwsSigner implement that as an extension?

Reply:
I think this makes sense if we want a generic request signing extension. If @lavignes wants to take that on in this PR series, that's fine, otherwise we could leave a TODO and do the refactor when we have a second signing extension that wants to do something similar.

Reply:
I opted for the TODO. It'd be trivial to expose it in a public header in the future.